File: tp1/tp.js
function guessNumber(max){
    let answer = Math.floor(Math.random()*max);
    let guess, stop = false;
    do {
        guess = prompt('Votre nombre : ');
        try {
            parseInt(guess);
            if(guess > answer){
                alert('Plus petit');
            } else if (guess < answer) {
                alert('Plus grand');
            }
        } catch ( e ) {
            if(guess != 'stop'){
                alert('Entrée invalide');
            } else {
                stop = true
            }
        }
    } while (guess != answer && !stop);

    if(stop){
        alert('Arrêt du programme (demande utilisateur)');
    } else {
        alert('Bravo vous avez trouvé');
    }
}

guessNumber(100);

File: tp1/tp.py
from random import random
import math

answer = math.floor(random()*1000)
guess = 0
stop = False
while not stop and int(guess) != answer :
    guess = input("Devinez le nombre : ")
    if guess.isdigit() :
        if int(guess) < answer :
            print("Plus grand")
        elif int(guess) > answer :
            print("Plus petit")
    elif guess != 'stop' :
        print("Entrée invalide")
    else :
        stop = True
 
if stop :
    print("Arrêt du programme (demande utilisateur)")
else :
    print("Bravo vous avez trouvé")
